### PR TITLE
go test: Reduce log output

### PR DESCRIPTION
**Pull request by efritz**
Reduce logs from worker and migration tests.

### CODE REVIEW

File: internal/database/migration/runner/logger.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+	"github.com/inconshreveable/log15"
+)
+
+// logger is the log15 root logger declared at the package level so it can be
+// replaced with a no-op logger in unrelated tests that need to run migrations.
+var logger = log15.Root()
+
+func init() {
+	if strings.HasSuffix(os.Args[0], ".test") || strings.Contains(os.Args[0], "/_test/") {
```
Comment (bobheadxi) [inline internal/database/migration/runner/logger.go:15]: If this change is specifically for buildkite, you can also just check for env BUILDKITE = true :) https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph+os.Getenv%28%22BUILDKITE%22%29&patternType=literal

Comment (efritz) [reply internal/database/migration/runner/logger.go:15]: It's actually super annoying in all local tests as well.

Comment (efritz) [reply internal/database/migration/runner/logger.go:15]: I'll update text here.

File: internal/database/migration/runner/logger.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		// This package is INCREDIBLY noisy and imported by everything that
+		// touches the database. We disable all logs during tests to save
+		// spewing useless logs into CI.
```
Comment (efritz) [inline internal/database/migration/runner/logger.go:18]:
```suggestion
		// This package is INCREDIBLY noisy and imported by basically any pacakge
		// that touches the database. We disable all logs during tests to save the noise
		// in local development as well as CI.
```

File: internal/workerutil/main_test.go [hunk 3]
```diff
@@ -9,9 +9,14 @@ import (
 )
 
 func TestMain(m *testing.M) {
+	// This package is INCREDIBLY noisy. We disable all logs during tests
+	// (regardless of the -v flag) to save spewing useless logs into CI.
```
Comment (efritz) [inline internal/workerutil/main_test.go:13]:
```suggestion
		// This package is INCREDIBLY noisy. We disable all logs during tests, regardless
		// of the `-v` flag, to save the noise in local development as well as CI.
```

File: internal/workerutil/logger.go [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package workerutil
+
+import "github.com/inconshreveable/log15"
+
+// logger is the log15 root logger declared at the package level so it can be
+// replaced with a no-op logger in unrelated tests that need to run migrations.
```
Comment (efritz) [inline internal/workerutil/logger.go:6]:
```suggestion
// replaced with a no-op logger in unrelated tests that use this code.
```

**Review by bobheadxi — APPROVED** [review]
thank you!!1!

**Pull request merged**